src/Compatibility/Microsoft.DotNet.ApiCompatibility/Rules/EnumsMustMatch.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+
+namespace Microsoft.DotNet.ApiCompatibility.Rules
+{
+    /// <summary>
+    /// This class implements a rule to check that the constant values for an enum's fields don't change.
+    /// </summary>
+    public class EnumsMustMatch : Rule
+    {
+        public override void Initialize(RuleRunnerContext context) => context.RegisterOnTypeSymbolAction(RunOnTypeSymbol);
+
+        private bool IsEnum(ITypeSymbol sym) => sym is not null && sym.TypeKind == TypeKind.Enum;
+
+        private void RunOnTypeSymbol(ITypeSymbol left, ITypeSymbol right, string leftName, string rightName, IList<CompatDifference> differences)
+        {
+            // Ensure that this rule only runs on enums.
+            if (!IsEnum(left) || !IsEnum(right))
+            {
+                return;
+            }
+
+            // Enum must be a named type to access its underlying type.
+            if (left is not INamedTypeSymbol l || right is not INamedTypeSymbol r)
+            {
+                return;
+            }
+
+            // Get enum's underlying type.
+            if (l.EnumUnderlyingType is not INamedTypeSymbol leftType || r.EnumUnderlyingType is not INamedTypeSymbol rightType)
```

Comment:
When would the EnumUnderlyingType not be a INamedTypeSymbol?

Reply:
`EnumUnderlyingType` is of type `INamedTypeSymbol?`. When nullable reference types are enabled, it gets flagged if I try to use it as an `INamedTypeSymbol`. Using `is not` lets me match against it and early return when it's anything other than `INamedTypeSymbol`.